src/middleware.ts

```diff
@@ -0,0 +1,21 @@
+import { NextRequest, NextResponse } from "next/server"
+
+import { DEFAULT_LOCALE, FAKE_LOCALE } from "./lib/constants"
+
+const PUBLIC_FILE = /\.(.*)$/
+
+export async function middleware(req: NextRequest) {
+  if (
+    req.nextUrl.pathname.startsWith("/_next") ||
+    req.nextUrl.pathname.includes("/api/") ||
+    PUBLIC_FILE.test(req.nextUrl.pathname)
+  ) {
+    return
+  }
+
+  if (req.nextUrl.locale === FAKE_LOCALE) {
+    return NextResponse.redirect(
+      new URL(`/${DEFAULT_LOCALE}${req.nextUrl.pathname}`, req.url)
+    )
+  }
```

Comment:
We seem to have lost the automatic redirect to the users preferred browser language, possibly with these lines here.

This was new with Next.js, but a user with a Spanish preference going to the homepage would get redirected to `/es` but now it's going to `/en`

Should we add logic here to get the users preferred language instead of just using `DEFAULT_LOCALE`?

Reply:
Also, seeing some `307 Temporary Redirect` when redirecting from "default"
```
/about 307 > 200
/about/ 307 > 301 > 200
```

Is there a way to use a 301 here instead?

Reply:
> Is there a way to use a 301 here instead?

Hmm as far as I can see there is no option to do a permanent redirect https://nextjs.org/docs/pages/api-reference/functions/next-response#redirect

Reply:
> We seem to have lost the automatic redirect to the users preferred browser language, possibly with these lines here.
> 
> This was new with Next.js, but a user with a Spanish preference going to the homepage would get redirected to `/es` but now it's going to `/en`
> 
> Should we add logic here to get the users preferred language instead of just using `DEFAULT_LOCALE`?

Oh good catch 😕 maybe that is why they were setting `localeDetection: false` in their example https://nextjs.org/docs/pages/building-your-application/routing/internationalization#prefixing-the-default-locale

That sucks...will play a bit to see if there is a way to handle this case.

Reply:
Ok, I tried several things like forcing the `localeDetection: true` but it didn't work. So, as a workaround, I'm implementing a manual detection of the user's locale inside the middleware. It works well in dev mode, will double check it here once it finish to build.

Reply:
@pettinarip 🤔 Hm... will test more to be 100% sure, but the latest commit df66026 seems to be working the way we want. And yeah, I'm not sure why, but testing the commit with `localeDetection:true` worked locally with `netlify dev` but not in the Netlify preview. I think this looks good if we don't discover any issues with it.

Reply:
Yeah, not feeling happy about that. Would prefer to use the built in solution 😞